File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& a, int n, vector<int>& b, int m) {
        
        vector<int> res(n + m);

        int i = 0, j = 0, id = 0;

        while(i < n && j < m){
            if(a[i] <= b[j]){
                res[id++] = a[i++];
            
            }
            else{
                res[id++] = b[j++];
               
            }
        }

        while(j < m){
            res[id++] = b[j++];
        }

        while(i < n){
            res[id++] = a[i++];
        }

        
        for(int k = 0; k < n + m; k++){
            a[k] = res[k];
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        i = 0
        j = 0
        id = 0
        res = []

        while i < m and j < n:
            if nums1[i] < nums2[j]:
                res.append(nums1[i])
                id += 1
                i += 1
            else:
                res.append(nums2[j])
                id += 1
                j += 1

        while j < n:
            res.append(nums2[j])
            id += 1
            j += 1

        while i < m:
            res.append(nums1[i])
            id += 1
            i += 1

        for k in range(m + n):
            nums1[k] = res[k]